crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -1023,6 +1062,42 @@ impl PaymentCreate {
             .change_context(errors::ApiErrorResponse::InternalServerError)?
             .map(Secret::new);
 
+        // Derivation of directly supplied Customer data in our Payment Create Request
+        let raw_customer_details = if request.name.is_some()
+            || request.email.is_some()
+            || request.phone.is_some()
+            || request.phone_country_code.is_some()
+        {
```

Comment:
in case of create, we need to store the customer details only in case where `customer_id` is not passed. This is because in cases where `customer_id` is passed, we fetch the customer details and update it in payment attempt in update trackers.